main/java/com/example/mvvm/fragment/ListFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.example.mvvm.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.mvvm.*
+import com.example.mvvm.adapter.UserAdapter
+import com.example.mvvm.database.AppDatabase
+import com.example.mvvm.databinding.FragmentListBinding
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
+
+class ListFragment : Fragment() {
+    private var _binding: FragmentListBinding? = null
+    private val binding get() = requireNotNull(_binding)
+
+    private val userRepository by inject<UserRepository>()
+    private val appDatabase by inject<AppDatabase>()
+
+    private val viewModel by viewModels<ListViewModel> {
+        object : ViewModelProvider.Factory {
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+                return ListViewModel(
+                    userRepository,
+                    appDatabase.userDao()
+                ) as T
+            }
+        }
+    }

```

Comment:
аналогично, как и с деталями, использовать делегат коина
```suggestion
    private val userRepository by inject<UserRepository>()
    private val appDatabase by inject<AppDatabase>()
    private val viewModel by viewModel<ListViewModel>()
```